classes/Kohana/Valid.php

```diff
@@ -346,13 +354,17 @@ public static function luhn($number)
 	 */
 	public static function phone($number, $lengths = NULL)
 	{
+		if ( ! is_string($number)){
```

Comment:
Worth the changelog mentioning that all the Valid:: methods for numeric strings (phone/card/alpha/digit etc) no longer validate int / float / values even if they would be valid in string form?

Reply:
**actually** I've just had a thought, although kohana validation is mostly used for POST, if it's used for e.g. entity validation (which ISTR we still do some places) then it may get integer etc input to the numeric validators.

So probably we do actually want things like `Valid::decimal`, `Valid::digit` at least to do something more like:
```
if (is_int($number) || is_float($number)) { 
  $number = string($number);
}
```

Which is a bit backwards really but will avoid risk of this breaking non-POST validators.

Sorry, realise that's the opposite of what I said yesterday.

Reply:
see 4c50074

Reply:
Ta